<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integration Test Human Non-small Cell Lung Carcinoma (NSCLC) • DSPWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Integration Test Human Non-small Cell Lung Carcinoma (NSCLC)">
<meta property="og:description" content="DSPWorkflow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DSPWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Integration_Test_Colon.html">Integration Test Human Colon</a>
    </li>
    <li>
      <a href="../articles/Integration_Test_Kidney.html">Integration Test Human Kidney</a>
    </li>
    <li>
      <a href="../articles/Integration_Test_Mouse_Thymus.html">Integration Test Mouse Thymus</a>
    </li>
    <li>
      <a href="../articles/Integration_Test_NSCLC.html">Integration Test Human Non-small Cell Lung Carcinoma (NSCLC)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NIDAP-Community/DSPWorkflow/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integration Test Human Non-small Cell Lung Carcinoma (NSCLC)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/NIDAP-Community/DSPWorkflow/blob/HEAD/vignettes/Integration_Test_NSCLC.Rmd" class="external-link"><code>vignettes/Integration_Test_NSCLC.Rmd</code></a></small>
      <div class="hidden name"><code>Integration_Test_NSCLC.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>echo <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: usethis</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rprojroot.r-lib.org/" class="external-link">rprojroot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://devtools.r-lib.org/reference/load_all.html" class="external-link">load_all</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Loading <span style="color: #0000BB;">DSPWorkflow</span></span></span></code></pre>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'GGally':</span></span>
<span><span class="co">##   method from   </span></span>
<span><span class="co">##   +.gg   ggplot2</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span></span>
<span><span class="co">##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span></span>
<span><span class="co">##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span></span>
<span><span class="co">##     union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: cowplot</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: dplyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     combine</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:BiocGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     combine, intersect, setdiff, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter, lag</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GeomxTools</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: NanoStringNCTools</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     first, rename</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: ggplot2</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'NanoStringNCTools'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     groups</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: ggforce</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: gridExtra</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'gridExtra'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     combine</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biobase':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     combine</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     combine</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: grid</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: gtable</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: knitr</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: patchwork</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'patchwork'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:cowplot':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     align_plots</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: reshape2</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Rtsne</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: scales</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SpatialDecon</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: tibble</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: tidyr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'tidyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:reshape2':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     smiths</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:S4Vectors':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: umap</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: magrittr</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'magrittr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:tidyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     extract</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: ComplexHeatmap</span></span></code></pre>
<pre><code><span><span class="co">## ========================================</span></span>
<span><span class="co">## ComplexHeatmap version 2.10.0</span></span>
<span><span class="co">## Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span><span class="co">## Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span><span class="co">## Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use it in published research, please cite:</span></span>
<span><span class="co">## Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span><span class="co">##   genomic data. Bioinformatics 2016.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The new InteractiveComplexHeatmap package can directly export static </span></span>
<span><span class="co">## complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## This message can be suppressed by:</span></span>
<span><span class="co">##   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span><span class="co">## ========================================</span></span></code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">root</span> <span class="op">&lt;-</span> <span class="fu">rprojroot</span><span class="fu">::</span><span class="fu"><a href="https://rprojroot.r-lib.org/reference/find_root_file.html" class="external-link">find_package_root_file</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>fig.width<span class="op">=</span><span class="fl">12</span>, fig.height<span class="op">=</span><span class="fl">8</span>, out.width <span class="op">=</span> <span class="st">'100%'</span><span class="op">)</span> </span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>root.dir <span class="op">=</span> <span class="va">root</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="r-markdown">R Markdown<a class="anchor" aria-label="anchor" href="#r-markdown"></a>
</h2>
<p>This runs the DSPworkflow package to completion using the Non Small Cell Lung Carcinoma Dataset:</p>
</div>
<div class="section level2">
<h2 id="study-design">1. Study Design:<a class="anchor" aria-label="anchor" href="#study-design"></a>
</h2>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure the working directory is the root package directory</span></span>
<span>  <span class="co"># Set paths for downloading dcc files</span></span>
<span>  <span class="va">downloads.path</span> <span class="op">&lt;-</span> <span class="fu">test_path</span><span class="op">(</span><span class="st">"fixtures/Human_NSCLC/downloaded/"</span><span class="op">)</span></span>
<span>  <span class="va">tar.file.name</span> <span class="op">&lt;-</span> <span class="st">"nsclc_dccs.tar.gz"</span></span>
<span>  <span class="va">full.tar.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">downloads.path</span>,<span class="va">tar.file.name</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Check if dcc files were previously downloaded</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">full.tar.path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    </span>
<span>    <span class="co"># Download dcc files and place in data folder</span></span>
<span>    <span class="va">data.url</span> <span class="op">&lt;-</span> <span class="st">"http://hpc.nih.gov/~CCBR/DSPWorkflow/nsclc_dccs.tar.gz"</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">data.url</span>, <span class="va">full.tar.path</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="va">full.tar.path</span>, exdir <span class="op">=</span> <span class="va">downloads.path</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="va">dcc.files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span>
<span>      <span class="va">downloads.path</span>,</span>
<span>      <span class="st">"dccs"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    pattern <span class="op">=</span> <span class="st">".dcc$"</span>,</span>
<span>    full.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    recursive <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span>    </span>
<span>  <span class="va">pkc.files</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">test_path</span><span class="op">(</span><span class="st">"fixtures/Human_NSCLC/DevCom_H_WTA_v1.0.pkc"</span><span class="op">)</span></span>
<span>  <span class="va">pheno.data.file</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">test_path</span><span class="op">(</span><span class="st">"fixtures/Human_NSCLC/NSCLC_annotation.xlsx"</span><span class="op">)</span>  </span>
<span>  </span>
<span>  <span class="va">sdesign.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/studyDesign.html">studyDesign</a></span><span class="op">(</span>dcc.files <span class="op">=</span> <span class="va">dcc.files</span>, </span>
<span>                              pkc.files <span class="op">=</span> <span class="va">pkc.files</span>,</span>
<span>                              pheno.data.file <span class="op">=</span> <span class="va">pheno.data.file</span>,</span>
<span>                              pheno.data.sheet <span class="op">=</span> <span class="st">"Template"</span>,</span>
<span>                              pheno.data.dcc.col.name <span class="op">=</span> <span class="st">"Sample_ID"</span>,</span>
<span>                              protocol.data.col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"aoi"</span>, <span class="st">"roi"</span><span class="op">)</span>,</span>
<span>                              experiment.data.col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"panel"</span><span class="op">)</span>,</span>
<span>                              slide.name.col <span class="op">=</span> <span class="st">"slide name"</span>, </span>
<span>                              class.col <span class="op">=</span> <span class="st">"class"</span>, </span>
<span>                              region.col <span class="op">=</span> <span class="st">"region"</span>, </span>
<span>                              segment.col <span class="op">=</span> <span class="st">"segment"</span>,</span>
<span>                              area.col <span class="op">=</span> <span class="st">"area"</span>,</span>
<span>                              nuclei.col <span class="op">=</span> <span class="st">"nuclei"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in studyDesign(dcc.files = dcc.files, pkc.files = pkc.files, pheno.data.file = pheno.data.file, : area is not found in the annotation and will not be considered</span></span></code></pre>
<pre><code><span><span class="co">## Warning in studyDesign(dcc.files = dcc.files, pkc.files = pkc.files, pheno.data.file = pheno.data.file, : nuclei is not found in the annotation and will not be considered</span></span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="co"># For creating fixture RDS</span></span>
<span>  <span class="va">create.rds</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="va">create.rds</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">study.design.human.nsclc</span> <span class="op">&lt;-</span> <span class="va">sdesign.list</span><span class="op">$</span><span class="va">object</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">study.design.human.nsclc</span>, file <span class="op">=</span> <span class="st">"tests/testthat/fixtures/Human_NSCLC/studyDesignHumanNSCLC.RDS"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">sdesign.list</span><span class="op">$</span><span class="va">sankey.plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Study%20Design-1.png" width="100%"></p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">"Created GeoMx Object\n\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Created GeoMx Object\n\n"</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="fu">pData</span><span class="op">(</span><span class="va">sdesign.list</span><span class="op">$</span><span class="va">object</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"slide_name"</span>,<span class="st">"class"</span>,<span class="st">"segment"</span><span class="op">)</span><span class="op">]</span>  </span></code></pre></div>
<pre><code><span><span class="co">##                                                slide_name   class    segment</span></span>
<span><span class="co">## GSM6573697_DSP-1012300141221-A-A02.dcc          Patient 1  cancer      tumor</span></span>
<span><span class="co">## GSM6573698_DSP-1012300141221-A-A03.dcc          Patient 1  cancer        TME</span></span>
<span><span class="co">## GSM6573699_DSP-1012300141221-A-A04.dcc          Patient 1  cancer      tumor</span></span>
<span><span class="co">## GSM6573700_DSP-1012300141221-A-A05.dcc          Patient 1  cancer      tumor</span></span>
<span><span class="co">## GSM6573701_DSP-1012300141221-A-A06.dcc          Patient 2  cancer      tumor</span></span>
<span><span class="co">## GSM6573702_DSP-1012300141221-A-A07.dcc          Patient 2  cancer      tumor</span></span>
<span><span class="co">## GSM6573703_DSP-1012300141221-A-A08.dcc          Patient 2  cancer        TME</span></span>
<span><span class="co">## GSM6573704_DSP-1012300141221-A-A09.dcc          Patient 2  cancer      tumor</span></span>
<span><span class="co">## GSM6573705_DSP-1012300141221-A-A10.dcc          Patient 3  cancer        TME</span></span>
<span><span class="co">## GSM6573706_DSP-1012300141221-A-A11.dcc          Patient 3  cancer      tumor</span></span>
<span><span class="co">## GSM6573707_DSP-1012300141221-A-A12.dcc          Patient 4  cancer      tumor</span></span>
<span><span class="co">## GSM6573708_DSP-1012300141221-A-B01.dcc          Patient 4  cancer      tumor</span></span>
<span><span class="co">## GSM6573709_DSP-1012300141221-A-B02.dcc          Patient 4  cancer        TME</span></span>
<span><span class="co">## GSM6573710_DSP-1012300141221-A-B03.dcc          Patient 4  cancer      tumor</span></span>
<span><span class="co">## GSM6573711_DSP-1012300141221-A-B04.dcc          Patient 5  cancer      tumor</span></span>
<span><span class="co">## GSM6573712_DSP-1012300141221-A-B05.dcc          Patient 5  cancer immune TME</span></span>
<span><span class="co">## GSM6573713_DSP-1012300141221-A-B06.dcc          Patient 5  cancer        TME</span></span>
<span><span class="co">## GSM6573714_DSP-1012300141221-A-B07.dcc          Patient 5  cancer      tumor</span></span>
<span><span class="co">## GSM6573715_DSP-1012300141221-A-B08.dcc          Patient 5  cancer immune TME</span></span>
<span><span class="co">## GSM6573716_DSP-1012300141221-A-B09.dcc          Patient 5  cancer      tumor</span></span>
<span><span class="co">## GSM6573717_DSP-1012300141221-A-B10.dcc          Patient 6  cancer      tumor</span></span>
<span><span class="co">## GSM6573718_DSP-1012300141221-A-B11.dcc          Patient 6  cancer        TME</span></span>
<span><span class="co">## GSM6573719_DSP-1012300141221-A-B12.dcc          Patient 6  cancer      tumor</span></span>
<span><span class="co">## GSM6573720_DSP-1012300141221-A-C01.dcc          Patient 7  cancer      tumor</span></span>
<span><span class="co">## GSM6573721_DSP-1012300141221-A-C02.dcc          Patient 7  cancer        TME</span></span>
<span><span class="co">## GSM6573722_DSP-1012300141221-A-C03.dcc          Patient 7  cancer      tumor</span></span>
<span><span class="co">## GSM6573723_DSP-1012300141221-A-C04.dcc          Patient 7  cancer      tumor</span></span>
<span><span class="co">## GSM6573724_DSP-1012300141221-A-C05.dcc          Patient 8  cancer      tumor</span></span>
<span><span class="co">## GSM6573725_DSP-1012300141221-A-C06.dcc          Patient 9  cancer        TME</span></span>
<span><span class="co">## GSM6573726_DSP-1012300141221-A-C07.dcc          Patient 9  cancer      tumor</span></span>
<span><span class="co">## GSM6573727_DSP-1012300141221-A-C08.dcc          Patient 9  cancer      tumor</span></span>
<span><span class="co">## GSM6573728_DSP-1012300141221-A-C09.dcc          Patient 9  cancer      tumor</span></span>
<span><span class="co">## GSM6573729_DSP-1012300141221-A-C10.dcc         Patient 10  cancer      tumor</span></span>
<span><span class="co">## GSM6573730_DSP-1012300141221-A-C11.dcc         Patient 10  cancer        TME</span></span>
<span><span class="co">## GSM6573731_DSP-1012300141221-A-C12.dcc         Patient 10  cancer      tumor</span></span>
<span><span class="co">## GSM6573732_DSP-1012300141221-A-D01.dcc         Patient 11  cancer      tumor</span></span>
<span><span class="co">## GSM6573733_DSP-1012300141221-A-D02.dcc         Patient 11  cancer      tumor</span></span>
<span><span class="co">## GSM6573734_DSP-1012300141221-A-D03.dcc         Patient 12  cancer      tumor</span></span>
<span><span class="co">## GSM6573735_DSP-1012300141221-A-D04.dcc         Patient 12  cancer      tumor</span></span>
<span><span class="co">## GSM6573736_DSP-1012300141221-A-D05.dcc         Patient 12  cancer immune TME</span></span>
<span><span class="co">## GSM6573737_DSP-1012300141221-A-D06.dcc         Patient 12  cancer      tumor</span></span>
<span><span class="co">## GSM6573738_DSP-1012300141221-A-D07.dcc         Patient 13  cancer      tumor</span></span>
<span><span class="co">## GSM6573739_DSP-1012300141221-A-D08.dcc         Patient 13  cancer        TME</span></span>
<span><span class="co">## GSM6573740_DSP-1012300141221-A-D09.dcc         Patient 13  cancer      tumor</span></span>
<span><span class="co">## GSM6573741_DSP-1012300141221-A-D10.dcc         Patient 14  cancer      tumor</span></span>
<span><span class="co">## GSM6573742_DSP-1012300141221-A-D11.dcc         Patient 14  cancer immune TME</span></span>
<span><span class="co">## GSM6573743_DSP-1012300141221-A-D12.dcc         Patient 14  cancer      tumor</span></span>
<span><span class="co">## GSM6573744_DSP-1012300141221-A-E01.dcc         Patient 14  cancer      tumor</span></span>
<span><span class="co">## GSM6573745_DSP-1012300141221-A-E02.dcc         Patient 15  cancer      tumor</span></span>
<span><span class="co">## GSM6573746_DSP-1012300141221-A-E03.dcc         Patient 15  cancer immune TME</span></span>
<span><span class="co">## GSM6573747_DSP-1012300141221-A-E04.dcc         Patient 15  cancer        TME</span></span>
<span><span class="co">## GSM6573748_DSP-1012300141221-A-E05.dcc         Patient 16  cancer      tumor</span></span>
<span><span class="co">## GSM6573749_DSP-1012300141221-A-E06.dcc         Patient 16  cancer        TME</span></span>
<span><span class="co">## GSM6573750_DSP-1012300141221-A-E07.dcc         Patient 16  cancer      tumor</span></span>
<span><span class="co">## GSM6573751_DSP-1012300141221-A-E08.dcc         Patient 17  cancer      tumor</span></span>
<span><span class="co">## GSM6573752_DSP-1012300141221-A-E09.dcc         Patient 17  cancer        TME</span></span>
<span><span class="co">## GSM6573753_DSP-1012300141221-A-E10.dcc         Patient 17  cancer      tumor</span></span>
<span><span class="co">## GSM6573754_DSP-1012300141221-A-E11.dcc         Patient 18  cancer        TME</span></span>
<span><span class="co">## GSM6573755_DSP-1012300141221-A-E12.dcc         Patient 19  cancer      tumor</span></span>
<span><span class="co">## GSM6573756_DSP-1012300141221-A-F01.dcc         Patient 19  cancer      tumor</span></span>
<span><span class="co">## GSM6573757_DSP-1012300141221-A-F02.dcc         Patient 20  cancer      tumor</span></span>
<span><span class="co">## GSM6573758_DSP-1012300141221-A-F03.dcc         Patient 20  cancer immune TME</span></span>
<span><span class="co">## GSM6573759_DSP-1012300141221-A-F04.dcc         Patient 20  cancer        TME</span></span>
<span><span class="co">## GSM6573760_DSP-1012300141221-A-F05.dcc         Patient 20  cancer      tumor</span></span>
<span><span class="co">## GSM6573761_DSP-1012300141221-A-F06.dcc         Patient 20  cancer      tumor</span></span>
<span><span class="co">## GSM6573762_DSP-1012300141221-A-F07.dcc         Patient 15  cancer      tumor</span></span>
<span><span class="co">## GSM6573763_DSP-1012300141221-A-F08.dcc         Patient 15  cancer immune TME</span></span>
<span><span class="co">## GSM6573764_DSP-1012300141221-A-F09.dcc         Patient 19  cancer immune TME</span></span>
<span><span class="co">## GSM6573765_DSP-1012300141221-A-F10.dcc         Patient 19  cancer immune TME</span></span>
<span><span class="co">## GSM6573766_DSP-1012300141221-A-F11.dcc          Patient 3  cancer      tumor</span></span>
<span><span class="co">## GSM6573767_DSP-1012300141221-A-F12.dcc          Patient 3  cancer      tumor</span></span>
<span><span class="co">## GSM6573768_DSP-1012300141221-A-G01.dcc         Patient 11  cancer        TME</span></span>
<span><span class="co">## GSM6573769_DSP-1012300141221-A-G03.dcc         Patient 12  cancer immune TME</span></span>
<span><span class="co">## GSM6573770_DSP-1012300141221-A-G05.dcc         Patient 15  cancer        TME</span></span>
<span><span class="co">## GSM6573771_DSP-1012300141221-A-G07.dcc         Patient 22  cancer      tumor</span></span>
<span><span class="co">## GSM6573772_DSP-1012300141221-A-G08.dcc         Patient 22  cancer        TME</span></span>
<span><span class="co">## GSM6573773_DSP-1012300141221-A-G09.dcc         Patient 21  cancer      tumor</span></span>
<span><span class="co">## GSM6573774_DSP-1012300141221-A-G10.dcc         Patient 21  cancer      tumor</span></span>
<span><span class="co">## GSM6573775_DSP-1012300141221-A-G11.dcc         Patient 19  cancer        TME</span></span>
<span><span class="co">## GSM6573776_DSP-1012300141221-A-G12.dcc         Patient 18  cancer      tumor</span></span>
<span><span class="co">## GSM6573777_DSP-1012300141221-A-H01.dcc         Patient 18  cancer immune TME</span></span>
<span><span class="co">## GSM6573778_DSP-1012300141221-A-H02.dcc         Patient 18  cancer      tumor</span></span>
<span><span class="co">## GSM6573779_DSP-1012300141221-A-H03.dcc         Patient 24  cancer      tumor</span></span>
<span><span class="co">## GSM6573780_DSP-1012300141221-A-H04.dcc         Patient 24  cancer      tumor</span></span>
<span><span class="co">## GSM6573781_DSP-1012300141221-A-H05.dcc         Patient 24  cancer immune TME</span></span>
<span><span class="co">## GSM6573782_DSP-1012300141221-A-H07.dcc         Patient 12  cancer immune TME</span></span>
<span><span class="co">## GSM6573783_DSP-1012300141221-A-H08.dcc         Patient 24  cancer immune TME</span></span>
<span><span class="co">## GSM6573784_DSP-1012300141221-A-H09.dcc         Patient 26  cancer      tumor</span></span>
<span><span class="co">## GSM6573785_DSP-1012300141221-A-H10.dcc         Patient 26  cancer immune TME</span></span>
<span><span class="co">## GSM6573786_DSP-1012300141221-A-H11.dcc         Patient 27  cancer        TME</span></span>
<span><span class="co">## GSM6573787_DSP-1012300141221-A-H12.dcc         Patient 40  cancer      tumor</span></span>
<span><span class="co">## GSM6573788_DSP-1012310141221-B-A02.dcc         Patient 29  cancer      tumor</span></span>
<span><span class="co">## GSM6573789_DSP-1012310141221-B-A03.dcc         Patient 29  cancer immune TME</span></span>
<span><span class="co">## GSM6573790_DSP-1012310141221-B-A04.dcc         Patient 34  cancer      tumor</span></span>
<span><span class="co">## GSM6573791_DSP-1012310141221-B-A05.dcc         Patient 34  cancer      tumor</span></span>
<span><span class="co">## GSM6573792_DSP-1012310141221-B-A06.dcc         Patient 32  cancer immune TME</span></span>
<span><span class="co">## GSM6573793_DSP-1012310141221-B-A07.dcc         Patient 32  cancer      tumor</span></span>
<span><span class="co">## GSM6573794_DSP-1012310141221-B-A08.dcc         Patient 31  cancer      tumor</span></span>
<span><span class="co">## GSM6573795_DSP-1012310141221-B-A09.dcc         Patient 31  cancer immune TME</span></span>
<span><span class="co">## GSM6573796_DSP-1012310141221-B-A10.dcc         Patient 43  cancer      tumor</span></span>
<span><span class="co">## GSM6573797_DSP-1012310141221-B-A11.dcc         Patient 43  cancer immune TME</span></span>
<span><span class="co">## GSM6573798_DSP-1012310141221-B-A12.dcc         Patient 30  cancer      tumor</span></span>
<span><span class="co">## GSM6573799_DSP-1012310141221-B-B01.dcc         Patient 30  cancer immune TME</span></span>
<span><span class="co">## GSM6573800_DSP-1012310141221-B-B02.dcc         Patient 35  cancer      tumor</span></span>
<span><span class="co">## GSM6573801_DSP-1012310141221-B-B03.dcc         Patient 35  cancer immune TME</span></span>
<span><span class="co">## GSM6573802_DSP-1012310141221-B-B04.dcc         Patient 35  cancer      tumor</span></span>
<span><span class="co">## GSM6573803_DSP-1012310141221-B-B05.dcc         Patient 35  cancer immune TME</span></span>
<span><span class="co">## GSM6573804_DSP-1012310141221-B-B06.dcc         Patient 37  cancer      tumor</span></span>
<span><span class="co">## GSM6573805_DSP-1012310141221-B-B07.dcc         Patient 37  cancer      tumor</span></span>
<span><span class="co">## GSM6573806_DSP-1012310141221-B-B08.dcc         Patient 40  cancer      tumor</span></span>
<span><span class="co">## GSM6573807_DSP-1012310141221-B-B09.dcc         Patient 40  cancer immune TME</span></span>
<span><span class="co">## GSM6573808_DSP-1012310141221-B-B11.dcc         Patient 44  cancer      tumor</span></span>
<span><span class="co">## GSM6573809_DSP-1012310141221-B-B12.dcc         Patient 44  cancer      tumor</span></span>
<span><span class="co">## GSM6573810_DSP-1012310141221-B-C01.dcc Non-tumor brain #1 control    control</span></span>
<span><span class="co">## GSM6573811_DSP-1012310141221-B-C02.dcc Non-tumor brain #2 control    control</span></span>
<span><span class="co">## GSM6573812_DSP-1012310141221-B-C04.dcc Non-tumor brain #3 control    control</span></span>
<span><span class="co">## GSM6573813_DSP-1012310141221-B-C05.dcc Non-tumor brain #4 control    control</span></span>
<span><span class="co">## GSM6573814_DSP-1012310141221-B-C06.dcc Non-tumor brain #5 control    control</span></span>
<span><span class="co">## GSM6573815_DSP-1012310141221-B-C07.dcc Non-tumor brain #6 control    control</span></span>
<span><span class="co">## GSM6573816_DSP-1012310141221-B-C08.dcc Non-tumor brain #7 control    control</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="qc-preprocessing">2. QC Preprocessing:<a class="anchor" aria-label="anchor" href="#qc-preprocessing"></a>
</h2>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc.output</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/qcProc.html">qcProc</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sdesign.list</span><span class="op">$</span><span class="va">object</span>,</span>
<span>                        min.segment.reads <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>                        percent.trimmed <span class="op">=</span> <span class="fl">80</span>,    </span>
<span>                        percent.stitched <span class="op">=</span> <span class="fl">80</span>,   </span>
<span>                        percent.aligned <span class="op">=</span> <span class="fl">80</span>,    </span>
<span>                        percent.saturation <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                        min.negative.count <span class="op">=</span> <span class="fl">10</span>,   </span>
<span>                        max.ntc.count <span class="op">=</span> <span class="cn">NULL</span>,     </span>
<span>                        min.nuclei <span class="op">=</span> <span class="cn">NULL</span>,         </span>
<span>                        min.area <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                        print.plots <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in qcProc(object = sdesign.list$object, min.segment.reads = 1000, : NTC, nuclei, area not found in the annotation, max.ntc.count, min.nuclei, min.area will not be considered</span></span></code></pre>
<p><img src="Integration_Test_NSCLC_files/figure-html/QC%20Preprocessing-1.png" width="100%"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Table: QC Summary for each Segment</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## |              | Pass| Warning|</span></span>
<span><span class="co">## |:-------------|----:|-------:|</span></span>
<span><span class="co">## |LowReads      |  120|       0|</span></span>
<span><span class="co">## |LowTrimmed    |  120|       0|</span></span>
<span><span class="co">## |LowStitched   |  120|       0|</span></span>
<span><span class="co">## |LowAligned    |  120|       0|</span></span>
<span><span class="co">## |LowSaturation |  120|       0|</span></span>
<span><span class="co">## |LowNegatives  |  114|       6|</span></span>
<span><span class="co">## |TOTAL FLAGS   |  114|       6|</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Table: Summary for Segment QC Removal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## |         | # Before Removal| # After Removal|</span></span>
<span><span class="co">## |:--------|----------------:|---------------:|</span></span>
<span><span class="co">## |Features |            18834|           18834|</span></span>
<span><span class="co">## |Samples  |              120|             114|</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Table: Summary for Probe QC Calls (Grubb's Outlier Test)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## | Passed| Global| Local|</span></span>
<span><span class="co">## |------:|------:|-----:|</span></span>
<span><span class="co">## |  18831|      1|     2|</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Table: Summary for Probe QC Removal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## |         | # Before Collapsing| # After Collapsing|</span></span>
<span><span class="co">## |:--------|-------------------:|------------------:|</span></span>
<span><span class="co">## |Features |               18834|              18833|</span></span>
<span><span class="co">## |Samples  |                 114|                114|</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Table: Summary for Gene-level Counts</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## |         | # Before Collapsing| # After Collapsing|</span></span>
<span><span class="co">## |:--------|-------------------:|------------------:|</span></span>
<span><span class="co">## |Features |               18833|              18695|</span></span>
<span><span class="co">## |Samples  |                 114|                114|</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">qc.output</span><span class="op">$</span><span class="va">segments.qc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="co"># For creating a fixture RDS</span></span>
<span>    <span class="va">create.rds</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="va">create.rds</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="va">qc.human.nsclc</span> <span class="op">&lt;-</span> <span class="va">qc.output</span><span class="op">$</span><span class="va">object</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">qc.human.nsclc</span>, file <span class="op">=</span> <span class="st">"tests/testthat/fixtures/Human_NSCLC/qcHumanNSCLC.RDS"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="filtering">3. Filtering:<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h2>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">goi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PDCD1"</span>, <span class="st">"CD274"</span>, <span class="st">"IFNG"</span>, <span class="st">"CD8A"</span>, <span class="st">"CD68"</span>, <span class="st">"EPCAM"</span>, <span class="st">"KRT18"</span>, <span class="st">"NPHS1"</span>, <span class="st">"NPHS2"</span>, <span class="st">"CALB1"</span>, <span class="st">"CLDN8"</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">filtering.output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filtering.html">filtering</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">qc.output</span><span class="op">$</span><span class="va">object</span>,</span>
<span>                            loq.cutoff <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                            loq.min <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                            cut.segment <span class="op">=</span> <span class="fl">.05</span>, </span>
<span>                            goi <span class="op">=</span> <span class="va">goi</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">filtering.output</span><span class="op">$</span><span class="va">`stacked.bar.plot`</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Filtering-1.png" width="100%"></p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">filtering.output</span><span class="op">$</span><span class="va">`tab`</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## |       | cancer| control|</span></span>
<span><span class="co">## |:------|------:|-------:|</span></span>
<span><span class="co">## |&lt;1%    |      0|       0|</span></span>
<span><span class="co">## |1-5%   |      1|       0|</span></span>
<span><span class="co">## |5-10%  |      2|       0|</span></span>
<span><span class="co">## |10-15% |      5|       1|</span></span>
<span><span class="co">## |&gt;15%   |     99|       6|</span></span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">filtering.output</span><span class="op">$</span><span class="va">`sankey.plot`</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Filtering-2.png" width="100%"></p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">filtering.output</span><span class="op">$</span><span class="va">`genes.detected.plot`</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Filtering-3.png" width="100%"></p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">create.rds</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="va">create.rds</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">filtering.human.nsclc</span> <span class="op">&lt;-</span> <span class="va">filtering.output</span><span class="op">$</span><span class="va">object</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">filtering.human.nsclc</span>, file <span class="op">=</span> <span class="st">"tests/testthat/fixtures/Human_NSCLC/filteringHumanNSCLC.RDS"</span><span class="op">)</span></span>
<span>  <span class="op">}</span> </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="normalization">4. Normalization:<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">q3.normalization.output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/geomxNorm.html">geomxNorm</a></span><span class="op">(</span></span>
<span>                                  object <span class="op">=</span> <span class="va">filtering.output</span><span class="op">$</span><span class="va">object</span>, </span>
<span>                                  norm <span class="op">=</span> <span class="st">"q3"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using Segment, Annotation as id variables</span></span></code></pre>
<pre><code><span><span class="co">## No id variables; using all as measure variables</span></span>
<span><span class="co">## No id variables; using all as measure variables</span></span></code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">multi.plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Normalization-1.png" width="100%"></p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">boxplot.raw</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Normalization-2.png" width="100%"></p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">boxplot.norm</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Normalization-3.png" width="100%"></p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">neg.normalization.output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/geomxNorm.html">geomxNorm</a></span><span class="op">(</span></span>
<span>                                  object <span class="op">=</span> <span class="va">filtering.output</span><span class="op">$</span><span class="va">object</span>, </span>
<span>                                  norm <span class="op">=</span> <span class="st">"neg"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using Segment, Annotation as id variables</span></span>
<span><span class="co">## No id variables; using all as measure variables</span></span>
<span><span class="co">## No id variables; using all as measure variables</span></span></code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">neg.normalization.output</span><span class="op">$</span><span class="va">multi.plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Normalization-4.png" width="100%"></p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">neg.normalization.output</span><span class="op">$</span><span class="va">boxplot.raw</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Normalization-5.png" width="100%"></p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">neg.normalization.output</span><span class="op">$</span><span class="va">boxplot.norm</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Normalization-6.png" width="100%"></p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">create.rds</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="va">create.rds</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="va">q3.normalization.human.nsclc</span> <span class="op">&lt;-</span> <span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">object</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">q3.normalization.human.nsclc</span>, file <span class="op">=</span> <span class="st">"tests/testthat/fixtures/Human_NSCLC/q3normalizationHumanNSCLC.RDS"</span><span class="op">)</span></span>
<span>      <span class="va">neg.normalization.human.nsclc</span> <span class="op">&lt;-</span> <span class="va">neg.normalization.output</span><span class="op">$</span><span class="va">object</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">neg.normalization.human.nsclc</span>, file <span class="op">=</span> <span class="st">"tests/testthat/fixtures/Human_NSCLC/negnormalizationHumanNSCLC.RDS"</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="unsupervised-analysis">5. Unsupervised Analysis:<a class="anchor" aria-label="anchor" href="#unsupervised-analysis"></a>
</h2>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="co">#Test Unsupervised Analysis:</span></span>
<span>    <span class="va">unsupervised.output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dimReduct.html">dimReduct</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">object</span>,</span>
<span>                        point.size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                        point.alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                        color.variable1 <span class="op">=</span> <span class="st">"region"</span>,</span>
<span>                        shape.variable <span class="op">=</span> <span class="st">"class"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## using q_norm in the dimensional reductions</span></span></code></pre>
<pre><code><span><span class="co">## adding in the phenoData PCA, tSNE, and UMAP coordinates</span></span></code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">unsupervised.output</span><span class="op">$</span><span class="va">plot</span><span class="op">$</span><span class="va">PCA</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Unsupervised%20Analysis-1.png" width="100%"></p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">unsupervised.output</span><span class="op">$</span><span class="va">plot</span><span class="op">$</span><span class="va">tSNE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Unsupervised%20Analysis-2.png" width="100%"></p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">unsupervised.output</span><span class="op">$</span><span class="va">plot</span><span class="op">$</span><span class="va">UMAP</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Unsupervised%20Analysis-3.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="clustering-high-cv-genes-and-heatmap">6. Clustering high CV Genes and Heatmap:<a class="anchor" aria-label="anchor" href="#clustering-high-cv-genes-and-heatmap"></a>
</h2>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">heatmap.output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/heatMap.html">heatMap</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">unsupervised.output</span><span class="op">$</span><span class="va">object</span>, </span>
<span>                        ngenes <span class="op">=</span> <span class="fl">200</span>, </span>
<span>                        scale.by.row.or.col <span class="op">=</span> <span class="st">"row"</span>, </span>
<span>                        show.rownames <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                        show.colnames <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                        clustering.method <span class="op">=</span> <span class="st">"average"</span>, </span>
<span>                        cluster.rows <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                        cluster.cols <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                        clustering.distance.rows <span class="op">=</span> <span class="st">"correlation"</span>, </span>
<span>                        clustering.distance.cols <span class="op">=</span> <span class="st">"correlation"</span>, </span>
<span>                        annotation.row <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>                        annotation.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"class"</span>, <span class="st">"segment"</span>, <span class="st">"region"</span><span class="op">)</span>, </span>
<span>                        breaks.by.values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">0.05</span><span class="op">)</span>, </span>
<span>                        heatmap.color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">120</span><span class="op">)</span>, </span>
<span>                        norm.method <span class="op">=</span> <span class="st">"quant"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: It not suggested to both set `scale` and `breaks`. It makes the</span></span>
<span><span class="co">## function confused.</span></span></code></pre>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">heatmap.output</span><span class="op">$</span><span class="va">plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Clustering%20high%20CV%20Genes-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="differential-expression-analysis">7. Differential Expression Analysis:<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h2>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">goi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ALDOC"</span>, <span class="st">"NCAM1"</span>, <span class="st">"SNAP25"</span>, <span class="st">"VIM"</span>, <span class="st">"COL3A1"</span>, </span>
<span>             <span class="st">"COL1A1"</span>, <span class="st">"COL4A1"</span>, <span class="st">"FN1"</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">object</span> <span class="op">&lt;-</span> <span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">object</span></span>
<span>    <span class="va">object</span> <span class="op">&lt;-</span> <span class="va">object</span><span class="op">[</span><span class="va">goi</span>,<span class="op">]</span></span>
<span>    <span class="va">Gene</span> <span class="op">&lt;-</span> <span class="va">Subset</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span>    </span>
<span>    <span class="co">#First analysis:</span></span>
<span>    <span class="va">reslist.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffExpr.html">diffExpr</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span>, </span>
<span>                          analysis.type <span class="op">=</span> <span class="st">"Within Groups"</span>, </span>
<span>                          region.col <span class="op">=</span> <span class="st">"segment"</span>, </span>
<span>                          regions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tumor"</span>, <span class="st">"TME"</span><span class="op">)</span>, </span>
<span>                          group.col <span class="op">=</span> <span class="st">"class"</span>, </span>
<span>                          groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cancer"</span><span class="op">)</span>, </span>
<span>                          n.cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "The number of cores selected is greater than the number of available cores, reducing number of cores to maximum of 2"</span></span>
<span><span class="co">## At least one of the regions within the Region Column was not selected</span></span>
<span><span class="co">##             and is excluded: immune TME, control</span></span></code></pre>
<pre><code><span><span class="co">## Running Within Group Analysis between Regions</span></span>
<span><span class="co">## Number of regions in group cancer: 2</span></span></code></pre>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu">grid.draw</span><span class="op">(</span><span class="va">reslist.1</span><span class="op">$</span><span class="va">sample_table</span><span class="op">)</span></span>
<span>    <span class="fu">grid.newpage</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="fu">grid.draw</span><span class="op">(</span><span class="va">reslist.1</span><span class="op">$</span><span class="va">summary_table</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">lfc_col1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.1</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"logFC"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.1</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="va">pval_col1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.1</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"_pval"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.1</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>    </span>
<span>    <span class="va">lfc.1</span> <span class="op">&lt;-</span> <span class="va">reslist.1</span><span class="op">$</span><span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene</span> <span class="op">==</span> <span class="st">"VIM"</span> <span class="op">&amp;</span> <span class="va">Subset</span> <span class="op">==</span> <span class="st">"cancer"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">select</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">lfc_col1</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="va">pval.1</span> <span class="op">&lt;-</span> <span class="va">reslist.1</span><span class="op">$</span><span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene</span> <span class="op">==</span> <span class="st">"VIM"</span> <span class="op">&amp;</span> <span class="va">Subset</span> <span class="op">==</span> <span class="st">"cancer"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">select</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">pval_col1</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"\n\nvalue of VIM Fold Change is:"</span>, <span class="va">lfc.1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## value of VIM Fold Change is:-2.09</span></span></code></pre>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"expected value is negative"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## expected value is negative</span></span></code></pre>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"\nvalue of VIM pval is:"</span>,<span class="va">pval.1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## value of VIM pval is:8.48e-07</span></span></code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"expected value is significant"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## expected value is significant</span></span></code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="co">#Second analysis:</span></span>
<span>    <span class="va">reslist.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffExpr.html">diffExpr</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span>, </span>
<span>                          analysis.type <span class="op">=</span> <span class="st">"Between Groups"</span>, </span>
<span>                          region.col <span class="op">=</span> <span class="st">"region"</span>, </span>
<span>                          regions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"brain"</span><span class="op">)</span>, </span>
<span>                          group.col <span class="op">=</span> <span class="st">"segment"</span>, </span>
<span>                          groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TME"</span>, <span class="st">"control"</span><span class="op">)</span>, </span>
<span>                          n.cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "The number of cores selected is greater than the number of available cores, reducing number of cores to maximum of 2"</span></span>
<span><span class="co">## At least one of the regions within the Region Column was not selected</span></span>
<span><span class="co">##             and is excluded: lung, lymph node</span></span></code></pre>
<pre><code><span><span class="co">## Running Between Group Analysis for Regions</span></span>
<span><span class="co">## Number of groups in region brain: 2</span></span></code></pre>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu">grid.draw</span><span class="op">(</span><span class="va">reslist.2</span><span class="op">$</span><span class="va">sample_table</span><span class="op">)</span></span>
<span>    <span class="fu">grid.newpage</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="fu">grid.draw</span><span class="op">(</span><span class="va">reslist.2</span><span class="op">$</span><span class="va">summary_table</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">lfc_col2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.2</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"logFC"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.2</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="va">pval_col2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.2</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"_pval"</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">reslist.2</span><span class="op">$</span><span class="va">result</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>    </span>
<span>    <span class="va">lfc.2</span> <span class="op">&lt;-</span> <span class="va">reslist.2</span><span class="op">$</span><span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene</span> <span class="op">==</span> <span class="st">"VIM"</span> <span class="op">&amp;</span> <span class="va">Subset</span> <span class="op">==</span> <span class="st">"brain"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">select</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">lfc_col2</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="va">pval.2</span> <span class="op">&lt;-</span> <span class="va">reslist.2</span><span class="op">$</span><span class="va">result</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Gene</span> <span class="op">==</span> <span class="st">"VIM"</span> <span class="op">&amp;</span> <span class="va">Subset</span> <span class="op">==</span> <span class="st">"brain"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">select</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">pval_col2</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"\n\nvalue of VIM Fold Change is:"</span>, <span class="va">lfc.2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## value of VIM Fold Change is:2.55</span></span></code></pre>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"expected value is positive"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## expected value is positive</span></span></code></pre>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"\nvalue of VIM pval is:"</span>,<span class="va">pval.2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## value of VIM pval is:9.79e-09</span></span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"expected value is significant"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## expected value is significant</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="volcano-plot">8. Volcano Plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h2>
<p>#This part is run on NIDAP.</p>
</div>
<div class="section level2">
<h2 id="violin-plot">9. Violin Plot<a class="anchor" aria-label="anchor" href="#violin-plot"></a>
</h2>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ALDOC"</span>, <span class="st">"NCAM1"</span>, <span class="st">"SNAP25"</span>, <span class="st">"VIM"</span>, <span class="st">"COL3A1"</span>, </span>
<span>             <span class="st">"COL1A1"</span>, <span class="st">"COL4A1"</span>, <span class="st">"FN1"</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">violin.plot.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/violinPlot.html">violinPlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">object</span>, </span>
<span>                                        expr.type <span class="op">=</span> <span class="st">"q_norm"</span>, </span>
<span>                                        genes <span class="op">=</span> <span class="va">genes</span>,</span>
<span>                                        group <span class="op">=</span> <span class="st">"region"</span>,</span>
<span>                                        facet.by <span class="op">=</span> <span class="st">"segment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] " not found and will not be displayed"</span></span></code></pre>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">violin.plot.test</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_Test_NSCLC_files/figure-html/Violin%20Plot-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="spatial-deconvolution">10. Spatial Deconvolution:<a class="anchor" aria-label="anchor" href="#spatial-deconvolution"></a>
</h2>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="va">ref.mtx</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu">test_path</span><span class="op">(</span><span class="st">"fixtures"</span>, <span class="st">"sample_spatial_deconv_mtx.csv"</span><span class="op">)</span>, </span>
<span>                       row.names<span class="op">=</span><span class="fl">1</span>, check.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ref.mtx</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">object</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1500</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>    <span class="va">ref.annot</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu">test_path</span><span class="op">(</span><span class="st">"fixtures"</span>, <span class="st">"ref_annot.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">spatial.output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialDeconvolution.html">spatialDeconvolution</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q3.normalization.output</span><span class="op">$</span><span class="va">object</span>, </span>
<span>                                           expr.type <span class="op">=</span> <span class="st">"q_norm"</span>, </span>
<span>                                           ref.mtx <span class="op">=</span> <span class="va">ref.mtx</span>, </span>
<span>                                           ref.annot <span class="op">=</span> <span class="va">ref.annot</span>,</span>
<span>                                           prof.mtx <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                                           use.custom.prof.mtx <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                           cell.id.col <span class="op">=</span> <span class="st">"CellID"</span>, </span>
<span>                                           celltype.col <span class="op">=</span> <span class="st">"LabeledCellType"</span>, </span>
<span>                                           group.by <span class="op">=</span> <span class="st">"segment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in create_profile_matrix(mtx = ref.mtx, cellAnnots = ref.annot, : not</span></span>
<span><span class="co">## all cellNameCol names are in count matrix; 5311 cells are missing</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Creating Atlas"</span></span>
<span><span class="co">## [1] "1 / 21 : cTEC"</span></span>
<span><span class="co">## [1] "2 / 21 : cTEC(cycling)"</span></span>
<span><span class="co">## [1] "3 / 21 : mTEC_I"</span></span>
<span><span class="co">## [1] "4 / 21 : mTEC_II"</span></span>
<span><span class="co">## [1] "5 / 21 : TEC_Cldn10"</span></span>
<span><span class="co">## [1] "6 / 21 : Epi(lung)"</span></span>
<span><span class="co">## [1] "7 / 21 : muscle"</span></span>
<span><span class="co">## [1] "8 / 21 : Fb_Postn"</span></span>
<span><span class="co">## [1] "9 / 21 : mTEC_III"</span></span>
<span><span class="co">## [1] "10 / 21 : Immune"</span></span>
<span><span class="co">## [1] "11 / 21 : TEC(neuro)_like_1"</span></span>
<span><span class="co">## [1] "12 / 21 : mTEC_IV(tuft)"</span></span>
<span><span class="co">## [1] "13 / 21 : Fb_Aldh1a2"</span></span>
<span><span class="co">## [1] "14 / 21 : Fb_Pi16"</span></span>
<span><span class="co">## [1] "15 / 21 : Endo"</span></span>
<span><span class="co">## [1] "16 / 21 : Mac"</span></span>
<span><span class="co">## [1] "17 / 21 : TEC(neuro)_like_2"</span></span>
<span><span class="co">## [1] "18 / 21 : Epi_Gcm2"</span></span>
<span><span class="co">## [1] "19 / 21 : VSMC"</span></span>
<span><span class="co">## [1] "20 / 21 : Epi_PAX8"</span></span>
<span><span class="co">## [1] "21 / 21 : Ery"</span></span></code></pre>
<pre><code><span><span class="co">## Using celltype as id variables</span></span></code></pre>
<pre><code><span><span class="co">## Warning in xtfrm.data.frame(x): cannot xtfrm data frames</span></span></code></pre>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">spatial.output</span><span class="op">$</span><span class="va">figures</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $abundance.heatmap</span></span></code></pre>
<p><img src="Integration_Test_NSCLC_files/figure-html/Spatial%20Deconvolution-1.png" width="100%"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $cell.profile.heatmap</span></span></code></pre>
<p><img src="Integration_Test_NSCLC_files/figure-html/Spatial%20Deconvolution-2.png" width="100%"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $composition.barplot</span></span></code></pre>
<p><img src="Integration_Test_NSCLC_files/figure-html/Spatial%20Deconvolution-3.png" width="100%"></p>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>    <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">"Spatial Deconvolution Done"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Spatial Deconvolution Done"</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Rui He, Maggie Cam, Ned Cauley, Jing Bian, Difei Wang, Chad Highfill, Aleksandra Michalowski.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
