<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Selects ROI/AOI segments and target probes based on quality control and then generates gene-level count data."><title>QcProc: Quality Control and Preprocessing — QcProc • DSPWorkflow</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="QcProc: Quality Control and Preprocessing — QcProc"><meta property="og:description" content="Selects ROI/AOI segments and target probes based on quality control and then generates gene-level count data."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DSPWorkflow</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Integration_Test_Colon.html">Integration_Test_Human_Colon</a>
    <a class="dropdown-item" href="../articles/Integration_Test_Kidney.html">Integration_Test_Kidney</a>
    <a class="dropdown-item" href="../articles/Integration_Test_Mouse_Thymus.html">Integration_Test_Kidney</a>
    <a class="dropdown-item" href="../articles/Integration_Test_NSCLC.html">Integration_Test_NSCLC</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>QcProc: Quality Control and Preprocessing</h1>
      
      <div class="d-none name"><code>QcProc.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Selects ROI/AOI segments and target probes based on quality control and then generates gene-level count data.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">QcProc</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  minSegmentReads <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  percentTrimmed <span class="op">=</span> <span class="fl">80</span>,</span>
<span>  percentStitched <span class="op">=</span> <span class="fl">80</span>,</span>
<span>  percentAligned <span class="op">=</span> <span class="fl">80</span>,</span>
<span>  percentSaturation <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  minNegativeCount <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  maxNTCCount <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minNuclei <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minArea <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minProbeRatio <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  outlierTestAlpha <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  percentFailGrubbs <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  removeLocalOutliers <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>object</dt>
<dd><p>object of class <em>NanoStringGeoMxSet</em> with raw data and metadata supplied</p></dd>


<dt>minSegmentReads</dt>
<dd><p>minimum number of reads</p></dd>


<dt>percentTrimmed</dt>
<dd><p>minimum % of reads trimmed</p></dd>


<dt>percentStitched</dt>
<dd><p>minimum % of reads stitched</p></dd>


<dt>percentAligned</dt>
<dd><p>minimum % of reads aligned</p></dd>


<dt>percentSaturation</dt>
<dd><p>minimum sequencing saturation</p></dd>


<dt>minNegativeCount</dt>
<dd><p>minimum negative control counts</p></dd>


<dt>maxNTCCount</dt>
<dd><p>maximum counts observed in NTC well</p></dd>


<dt>minNuclei</dt>
<dd><p>minimum # of nuclei estimated</p></dd>


<dt>minArea</dt>
<dd><p>minimum segment area</p></dd>


<dt>minProbeRatio</dt>
<dd><p>numeric between 0 and 1 to flag probes that have (geomean probe in all segments) / (geomean probes within target) less than or equal to this ratio</p></dd>


<dt>outlierTestAlpha</dt>
<dd><p>significance cut-off for Grubb's test to flag outlier probes</p></dd>


<dt>percentFailGrubbs</dt>
<dd><p>numeric to flag probes that fail Grubb's test in greater than or equal this percent of segments</p></dd>


<dt>removeLocalOutliers</dt>
<dd><p>boolean to determine if local outliers should be excluded from exprs matrix</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A list of two objects: QC-filtered <em>NanoStringGeoMxSet</em> object with gene-level targets as features and the <em>QCFlags</em> data frame appended to <em>protocolData</em> and QC plots</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>A probe is removed globally from the dataset if either it does not reach the <em>minProbeRatio</em> threshold or the probe is an outlier according to Grubb’s test in at least <em>percentFailGrubbs</em> of segments. A probe is removed locally (from a given segment) if the probe is an outlier in that segment and <em>removeLocalOtliers</em> is set to TRUE (default).</p>
<p>The count for any gene with multiple probes per segment is calculated as the geometric mean of those probes.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

